tsdb/chunkenc/chunk.go

```diff
@@ -56,11 +57,20 @@ type Appender interface {
 	Append(int64, float64)
 }
 
-// Iterator is a simple iterator that can only get the next value.
+// Iterator iterates over the samples of a time series.
 type Iterator interface {
+	// Seek advances the iterator forward to the sample with the timestamp t or first value after t.
+	// Seek has no effect if requested timestamp is the same or lower than the current iterator position.
+	// Seek returns false if there is no such sample with the timestamp equal or larger than t.
+	// Iterator is exhausted when the Seek returns false.
```

Comment:
Is that desired behavior? 
I mean does it make  sense to exhaust the iterator if the current time is 100 the iterator max time is 200 and you request Seek to 10? 
Logically I would think that exhausting the iterator would happen when you request 210 for example.

```suggestion
	// Iterator is exhausted when the requested Seek is higher than the iterator maximum timestamp.
```